### PR TITLE
Python virtual env support

### PR DESCRIPTION
**Pull request by mikapfl**
Actually, not a lot needed for python virtual env support.

The approach is now:
* one main venv
* for each run, write package versions from main venv into lockfile in the run folder, create new venv from that specification
* Specify requirements in main folder in requirements.txt file, main venv will be updated from that on each invocation.

Creating the main venv is done in .Rprofile, so can't be shared via piamenv, right?

### CODE REVIEW

File: R/updatePythonVirtualEnv.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#' updatePythonVirtualEnv
+#'
+#' Installs newly added requirements into the Python virtual environment.
+#'
+#' @param venv Path to Python virtual environment folder.
+#' @param requirements Path to a requirements.txt file containing the current requirements.
+#'
+#' @author Mika Pflüger
+#' @export
+updatePythonVirtualEnv <- function(venv = ".venv", requirements = "requirements.txt") {
```
Comment (pfuehrlich-pik) [inline R/updatePythonVirtualEnv.R:10]: You could consider calling this `updateVenv` to make the analogue to `updateRenv` clearer and to make it nicer to call interactively. Do you also plan to add some `make` targets for python venv to our lucode2 Makefile?

**Review by pfuehrlich-pik — APPROVED** [review]
Looks good. Just on a sidenote: For renv we also have an archive feature. If python dependencies reach a certain complexity level we should consider adding something similar for python venv as well.

File: R/updatePythonVirtualEnv.R (hunk 1 above)
Comment (mikapfl) [reply R/updatePythonVirtualEnv.R:10]: the lucode2 Makefile? So far, I'm only targeting models, not R packages, and models aren't managed with lucode2, no?

Comment (mikapfl) [reply R/updatePythonVirtualEnv.R:10]: I wasn't sure that people think of Python virtual environments if the read `venv` in an R context, that's why there is `Python` in all the function names.

Comment (pfuehrlich-pik) [reply R/updatePythonVirtualEnv.R:10]: No you're right, I meant the model Makefiles. Fair enough with python in the names.

**Pull request merged**